src/proxyp/Elements.cc

```diff
@@ -11,28 +11,53 @@
 #include "proxyp/Elements.h"
 #include "sbuf/Stream.h"
 
+#include <algorithm>
 #include <limits>
+#include <map>
 
-const ProxyProtocol::FieldMap ProxyProtocol::PseudoHeaderFields = {
-    { SBuf(":version"), ProxyProtocol::Two::htPseudoVersion },
-    { SBuf(":command"), ProxyProtocol::Two::htPseudoCommand },
-    { SBuf(":src_addr"), ProxyProtocol::Two::htPseudoSrcAddr },
-    { SBuf(":dst_addr"), ProxyProtocol::Two::htPseudoDstAddr },
-    { SBuf(":src_port"), ProxyProtocol::Two::htPseudoSrcPort },
-    { SBuf(":dst_port"), ProxyProtocol::Two::htPseudoDstPort }
+namespace ProxyProtocol {
+namespace Two {
+
+/// a mapping between pseudo header names and ids
+typedef std::map<SBuf, FieldType> FieldMap;
+static const FieldMap PseudoHeaderFields = {
+    { SBuf(":version"), htPseudoVersion },
+    { SBuf(":command"), htPseudoCommand },
+    { SBuf(":src_addr"), htPseudoSrcAddr },
+    { SBuf(":dst_addr"), htPseudoDstAddr },
+    { SBuf(":src_port"), htPseudoSrcPort },
+    { SBuf(":dst_port"), htPseudoDstPort }
 };
 
-namespace ProxyProtocol {
+} // namespace Two
+
 static Two::FieldType NameToFieldType(const SBuf &);
 static Two::FieldType IntegerToFieldType(const SBuf &);
+
 } // namespace ProxyProtocol
 
+
+const SBuf &
+ProxyProtocol::PseudoFieldTypeToFieldName(const Two::FieldType fieldType)
+{
+    // reverse PseudoHeaderFields so that we can look up names by FieldType
```

Comment:
"reverse" is the wrong term here. This is generating a new mapping, not changing the contents of PseudoHeaderFields.

Reply:
Addressed in 90e929a? Changed to "flip" in hope to guess what you consider the "right" term to be.

FWIW, "reverse" is a popular and meaningful term here, probably based on "reverse lookup" which is a well-known technical term. The code reverses the roles of map keys and map values to optimize reverse lookups. Lots of online discussions refer to this operation as reversing the map. I cannot find a better term for that operation ("transpose" is not it).

Reply:
The PseudoHeaderFields is a 2-dimensional container. It has two directions which can be reversed / flipped / swapped. The comment does not make it clear which is happening *and* implies PseudoHeaderFields (as the singular subject) is being changed somehow. So one has to understand both the code and mapping terminology to read the comment properly - which makes it confusing / broken.

The improvements that I can think of right now are:
 * name the static local pseudoHeaderFieldsByName and remove the comment.
  That makes the code speak for itself as being some way to initialize the _new_ mapping.

* have the comment say the code action, not the abstraction.
```suggestion
    // invert mapping from PseudoHeaderFields so that we can look up names by FieldType
```

Reply:
@eduard-bagdasaryan, after fixing the runtime problems, please use the comment wording and variable naming that Amos likes to address this change request.

Reply:
Fixed at 87f38f. The code around this comment was refactored and I removed it.